crates/env/src/call/call_builder.rs

```diff
@@ -192,7 +171,7 @@ where
 ///             .push_arg(true)
 ///             .push_arg(&[0x10; 32])
 ///     )
-///     .returns::<ReturnType<i32>>()
+///     .returns::<i32>()
```

Comment:
❤️ that users aren't exposed to `ReturnType<T>` anymore

Reply:
In the PR description you mention that _"the user can either omit the call to `returns::<R>()`"_. Doesn't looks like this is the case, if I remove this line the example fails to compile

Reply:
Oh it appears that is only the case if the arg list is not set  (which in this example it is) see https://github.com/paritytech/ink/blob/c0cccb74270f4ef9b92ea3dbde7a06b25f1bf091/crates/env/src/call/call_builder.rs#L438.

Here is an example which calls explcitily but with no args so that compiles: https://github.com/paritytech/ink/blob/378d83f3fc204e18dce6f3946e6a0f27e333d7d6/examples/proxy/lib.rs#L80

I'm not sure why it was already this way, it would make sense to either allow it to be `Unset` either way and default to `()`, or require it to be set in all cases. What do you think?